Currently the building gives:
 * QA Notice: Package has poor programming practices which may compile
 *            fine but exhibit random runtime failures.
 * ../../../src/solaris/native/java/net/Inet4AddressImpl.c:381: warning: implicit declaration of function 'gettimeofday'
../../../src/solaris/native/java/net/Inet6AddressImpl.c:511: warning: implicit declaration of function 'gettimeofday'
 * ../../../src/solaris/native/java/net/Inet4AddressImpl.c:381: warning: implicit declaration of function 'gettimeofday'
../../../src/solaris/native/java/net/Inet6AddressImpl.c:511: warning: implicit declaration of function 'gettimeofday'
 * ../../../src/solaris/native/java/net/Inet4AddressImpl.c:381: warning: implicit declaration of function 'gettimeofday'
../../../src/solaris/native/java/net/Inet6AddressImpl.c:511: warning: implicit declaration of function 'gettimeofday'
 * ../../../src/solaris/native/java/net/Inet4AddressImpl.c:381: warning: implicit declaration of function 'gettimeofday'
../../../src/solaris/native/java/net/Inet6AddressImpl.c:511: warning: implicit declaration of function 'gettimeofday'

This fixes those warnings by including the needed header.
Written by: betelgeuse@gentoo.org

Index: j2se/src/solaris/native/java/net/Inet4AddressImpl.c
===================================================================
--- openjdk.old/j2se/src/solaris/native/java/net/Inet4AddressImpl.c	(revision 235)
+++ openjdk/j2se/src/solaris/native/java/net/Inet4AddressImpl.c	(working copy)
@@ -24,6 +24,7 @@
  */
 
 #include <errno.h>
+#include <sys/time.h>
 #include <sys/types.h>
 #include <sys/socket.h>
 #include <netinet/in_systm.h>
Index: j2se/src/solaris/native/java/net/Inet6AddressImpl.c
===================================================================
--- openjdk.old/j2se/src/solaris/native/java/net/Inet6AddressImpl.c	(revision 235)
+++ openjdk/j2se/src/solaris/native/java/net/Inet6AddressImpl.c	(working copy)
@@ -24,6 +24,7 @@
  */
 
 #include <errno.h>
+#include <sys/time.h>
 #include <sys/types.h>
 #include <sys/socket.h>
 #include <netinet/in.h>
